parlai/utils/misc.py

```diff
@@ -311,6 +311,21 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
 
+class SimpleCounter:
+    """
+    Simple counter object.
+    """
+
+    def __init__(self, value=0):
+        self.val = value
+
+    def increment(self, value=1):
+        self.val += value
```

Comment:
so we don't need a lock here after all?

Reply:
Nope because the increment/access only happens on the main thread now, so we are good